account_invoice_portal_tag_hidden/views/account_portal_template.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="portal_my_invoices_inherit" name="Portal Tags Invoices Hidden" inherit_id="account.portal_my_invoices">
+        <xpath expr="//t[@t-call='portal.portal_table']/thead/tr/th[4]" position="attributes">
+            <attribute name="t-if">False</attribute>
+        </xpath>
+        <xpath expr="//t[@t-foreach='invoices']/tr/td[4]" position="attributes">
+            <attribute name="t-if">False</attribute>
+        </xpath>
+    </template>
+
+    <template id="portal_my_invoices_payment_inherit" name="Portal Tags Invoices Hidden" inherit_id="account_payment.portal_my_invoices_payment">
+        <th class="text-center" position="attributes">
+            <attribute name="t-if">False</attribute>
+        </th>
+        <td class="text-center" position="attributes">
+            <attribute name="t-if">False</attribute>
+        </td>
+    </template>
```

Comment:
I think this is equivalent to archiving `account_payment.portal_my_invoices_payment` view. Combining it with the other view inheritance, you will achive the desired behavior: hiding 4th and 5th column. Could you check?

With this approch inheriting from class attribute matching would be avoided